File: CSharpCodeBase/aiactions/ai_selectfood.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class ai_selectfood :ainode{
 // Select enemy using ai vision
 public void init(self, entity);
   AINode.init(self, entity);
 }})
 public void Visit(){
   if(this.status == NODE_READY  ){ 
     this.parent.selected_target = this.entity:GetPriorityFood();
     this.status = this.parent.selected_target != null  &&  NODE_SUCCESS  ||  NODE_FAILURE;
   }
   self:Sleep(0.5);
   return this.status;
 }
 }}

File: CSharpCodeBase/aiactions/ai_selectfood.lua
local Class = require 'utils.hump.class'
local AINode = require 'utils.ailib.ainode'

-- Select enemy using ai vision
local AI_SelectFood = Class({__includes=AINode, init=function(self, entity)
  AINode.init(self, entity)
end})

function AI_SelectFood:Visit()
  if self.status == NODE_READY then
    self.parent.selected_target = self.entity:GetPriorityFood()
    self.status = self.parent.selected_target ~= nil and NODE_SUCCESS or NODE_FAILURE
  end
  self:Sleep(0.5)
  return self.status
end

return AI_SelectFood